storage-proofs/src/stacked/params.rs

```diff
@@ -361,6 +479,71 @@ impl<H: Hasher> Labels<H> {
         self.labels.is_empty()
     }
 
+    pub fn labels_for_layer(&self, layer: usize) -> DiskStore<H::Domain> {
+        assert!(layer != 0, "Layer cannot be 0");
+        assert!(
+            layer <= self.layers(),
+            "Layer {} is not available (only {} layers available)",
+            layer,
+            self.layers()
+        );
+
+        let row_index = layer - 1;
+        let config = self.labels[row_index].clone();
+        assert!(config.size.is_some());
+
+        DiskStore::new_from_disk(config.size.unwrap(), &config).unwrap()
```

Comment:
This expression will panic inside of `seal_commit` if the merkle tree was deleted from the cache directory after `seal_pre_commit` returned.

@dignifiedquire - If we really really want to get the merkle cache in-use by lotus ASAP, then we could merge this PR as-is and create follow-up stories to deal with panics. Or, we could fix it now. Deferring to your judgement.

---

To repro, check out branch `mt-persistence-panic` and run the following:

```
cargo test --color=always --package filecoin-proofs --lib api::tests::test_seal_lifecycle_layer_missing -- --nocapture --exact --ignored
```